Sources/cte/Parser.swift

```diff
@@ -477,7 +537,7 @@ struct Parser {
     }
 
     mutating func consumeNewlines() {
-        while lexer.peek()?.kind == .newline {
+        while let token = lexer.peek()?.kind, token == .newline || token == .semicolon {
```

Comment:
Consume newlines should only consume newlines, we should have a second method `consumeTerminatorsAndNewlines` to specifically opt in to the behaviour.
For example in the parser code for the `fn` newlines are consumed where permitted to allow long functions to have their parameters split across multiple lines.
As it currently stands this compiler enforces code standards, dictating where newlines are actually permitted.